compiler/rustc_mir_transform/src/lib.rs

```diff
@@ -137,6 +142,65 @@ pub fn provide(providers: &mut Providers) {
     };
 }
 
+fn remap_mir_for_const_eval_select<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    mut body: Body<'tcx>,
+    context: hir::Constness,
+) -> Body<'tcx> {
+    for bb in body.basic_blocks.as_mut().iter_mut() {
+        let terminator = bb.terminator.as_mut().expect("invalid terminator");
+        // temporarily replace it with a dummy kind.
+        match std::mem::replace(&mut terminator.kind, TerminatorKind::Unreachable) {
```

Comment:
you could just mutably match on the terminator instead and `std::mem::take` the fields you want to move out of. You're only going to replace it with a `Call` terminator again anyway.

Reply:
Done